### ISSUE TITLE
Address search regression on Pelias Query

### ISSUE THREAD

**Issue by antoine-de** [pelias/pelias#690]
Hey team! 

I was using your awesome geocoding engine when I noticed something interesting.

#### Here's what I did :innocent:
I set up a pelias instance on france and I got strange regression on the last update (I nailed down the regression is on pelias/query [9.4 version](https://github.com/pelias/query/releases/tag/v9.4.0)

The latest version makes it more difficult to find addresses.

For example on a simple query (it's a french address)
`http ":4000/v1/search?text=19 Rue Denis Cordonnier Aniche"`

If I use the 9.3 version of pelias/query I got the address I want:
```               
 "label": "19 Rue Denis Cordonnier, Aniche, France",               
 "layer": "address",
```

and if I just upgrade the pelias/query to 9.4 version only the street is found (since the address lookup it's the street fallback that works):
```
"label": "Rue Denis Cordonnier, Aniche, France",
"layer": "street",
```

#### Here's what I think could be improved :trophy:

I think the [unit search feature](https://github.com/pelias/query/pull/77) can be improved by searching on the unit only if libpostal parsed one.

I think we should check that a unit has been parsed before calling `createAddressShould`:
https://github.com/pelias/query/blob/master/layout/AddressesUsingIdsQuery.js#L101

Just like how it's done in [StructuredFallbackQuery](https://github.com/pelias/query/blob/master/layout/StructuredFallbackQuery.js#L519-L520).

The fix seems quite easy and I'm willing to do it if you agree with the proposed solution. 

Note: I cannot reproduce it on the [compare site](http://pelias.github.io/compare/#/v1/search%3Ftext=19%20Rue%20Denis%20Cordonnier%20Aniche), but it might be because the pelias query version has not been upgraded yet. How can I know which pelias exact version has been deployed ?

**Comment by missinglink** [comment]
possibly related? https://github.com/pelias/pelias/issues/685

cc/ @sweco-semhul

**Comment by missinglink** [comment]
hi @antoine-de, thanks for the bug report.

If it's not too much work, could you please open a PR with the proposed code change and we can then discuss it on the PR?

regarding mapzen search, our production environment is running the `production` branch from each repo and our dev servers are usually running the `staging` branch.

**Comment by missinglink** [comment]
If you have some good test data from France that we are missing, please add it to the https://github.com/pelias/acceptance-tests repo.

These tests are run against the `prodbuild` environment before it is promoted to `production`, so it will ensure that we do not regress in the future.

Thanks!

**Comment by sweco-semhul** [comment]
@antoine-de yes it seems to be related to the changes made by me to add unit.

I say changing https://github.com/pelias/query/blob/master/layout/AddressesUsingIdsQuery.js#L101 to only search with unit if parsed would probably fix this.

**Comment by antoine-de** [comment]
I opened the [query's PR](https://github.com/pelias/query/pull/78)

I don't think it's related to #685 because the production branch is using query 9.1.0 and the staging 9.1.1 so there should be no problem in production.

For nice acceptance tests on french data I often use https://github.com/geocoders/geocoder-tester but it's not trivial to run them on Pelias (we've developed a [pelias proxy](https://github.com/QwantResearch/pelias-geocoder-tester-proxy) to run them).

If you want I can add some other tests on the `acceptance-tests` repository. Do I create a new file for french data ?
